src/frontends/sql/decoder_context.cc

```diff
@@ -38,7 +38,7 @@ const Operation &DecoderContext::MakeMergeOperation(
 
 const ir::Operation &DecoderContext::MakeTagTransformOperation(
     ir::Value transformed_value, absl::string_view rule_name,
-    std::vector<uint64_t> preconditions) {
+    absl::flat_hash_map<std::string, uint64_t> preconditions) {
```

Comment:
Since you are not making a copy of the map, should you change this to `const &`?

Reply:
OK, sounds good.